plugins/woocommerce/includes/data-stores/class-wc-order-data-store-cpt.php

```diff
@@ -1003,6 +1003,28 @@ protected function get_wp_query_args( $query_vars ) {
 	 * @return array|object
 	 */
 	public function query( $query_vars ) {
+		// Trigger doing_it_wrong() for query vars only supported in HPOS.
+		$hpos_only_query_vars   = array( 'meta_query', 'field_query' );
```

Comment:
If this array was customizable via filter hook or something, it seems like that would resolve the concern about 3rd party support... ?

Reply:
That's a good idea. I've added such filter. Thanks for the suggestion @coreymckrill!